SignalServiceKit/src/Storage/OWSStorage.m

```diff
@@ -79,7 +84,7 @@ - (void)readWriteWithBlock:(void (^)(YapDatabaseReadWriteTransaction *transactio
 {
     id<OWSDatabaseConnectionDelegate> delegate = self.delegate;
     OWSAssert(delegate);
-    OWSAssert(delegate.isDatabaseInitialized);
+    OWSAssert(delegate.areSyncRegistrationsComplete);
```

Comment:
Note that sometimes we gate on whether the _sync_ registrations are complete and sometimes we gate on whether _all_ registrations are complete.